### DISCUSSION TITLE
How to move forward with this spec?

### DISCUSSION THREAD

**Discussion by Emily-Jiang** [eclipse-ee4j/config#108] [General]
- [Programmatic API?](https://github.com/eclipse-ee4j/config/discussions/110)
- [Mutability?](https://github.com/eclipse-ee4j/config/discussions/109)
- [Injectable configuration?](https://github.com/eclipse-ee4j/config/discussions/111)
- [Dependent specifications?](#116)

**Comment by ljnelson** [comment]
Background for other readers:

"Configuration" is _enormous_. Everyone sees a different part of the elephant when they think about the subject.

The discussions linked in this discussion are an attempt to make people aware of the parts of the elephant.

* [Programmatic API](#110): You _could_ have a specification that only defines, say, file formats.  That would be an example of a specification that does _not_ define a programmatic API.  Or you could have a specification that dictates how to retrieve or otherwise acquire Java objects.  That would be an example of a specification that _does_ define a programmatic API.  Or you could have both of course.
* [Mutability](https://github.com/eclipse-ee4j/config/discussions/109): There have been many discussions in this group in the past around all aspects of mutability.  This may encompass (non-exhaustively!!) things like:
  * Can you change configuration from inside a running application?
  * Can you be notified of changes to configuration that occur outside a running application?
* [Injectable Configuration](#111): Regardless of any outcome from the other discussions, should the specification allow for `@Inject`-annotated points where configuration, in some shape or form, whatever that noun might mean, can be "magically" handed to a user?  If so, in which circumstances?
* [Dependent Specifications](#116): What Jakarta EE specifications should Jakarta Config depend on?

This discussion itself (#108) is *not* the place to comment or add content; please interact with the discussions linked from this master issue.  Committers: please feel free to edit this comment as needed.

**Comment by ljnelson** [comment]
I'm locking this conversation from further discussion, because all discussion should occur in the linked discussions.  Committers: feel free to unlock if for some reason this is not the proper thing to do.

**Comment by dmlloyd** [comment]
I've pinned this discussion; feel free to unpin it if that was the wrong thing to do. Perhaps we want to unpin the other topic as it is somewhat out of date.
